rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00244.rpgle

```diff
@@ -0,0 +1,12 @@
+     D £DBG_Str        S             2
+     D £C6JDSO         DS           512
+     D  £C6JO_TO                      2  0 DIM(10)
+     D $X              S              5  0
+     D §§IMPO          S              6  0
+     D T$AG_I          S             21  6 DIM(15)
+     C                   EVAL      $X=2
+     C                   EVAL      £C6JO_TO($X)=2
+     C                   EVAL      T$AG_I(£C6JO_TO($X))=
```

Comment:
What was wrong?

Reply:
In expressions like `T$AG_I(£C6JO_TO($X))` where the index expression is an array access it was detected as a `FunctionCall`. We already had a mechanism to replace it with an `ArrayAccessExpr` however it was only working at "first level" and was not replacing the `index` of the `index` of the `index` [...] like in line 9 of this test. Line 8 is present in order to avoid `index` being null and araising an error.

Reply:
The PR already contains a description related to the fix. I can reword it if it is not clear enough.

Reply:
By "high-level description" I mean using language that allows an RPGLE coder to understand what the PR fixes or improves. This is a new rule I'd like to introduce because I realized our PR descriptions are often too technical.

The rule could be as follows:
- Highlighting the RPGLE aspects is a must.
- Implementation details, especially when the changes are significant, are highly appreciated.

For this reason I have just changed the pr template:
https://github.com/smeup/jariko/blob/develop/docs/PULL_REQUEST_TEMPLATE.md